### PR TITLE
Dev/ep2019py37 unicode

### PR DESCRIPTION
**Pull request by fpierfed**
Python3.x already has unicode strings and so there is no need to encode() them. Also given that encoding a str in Python3.x outputs a bytes instance, not another str like in Python 2.x

This pull request builds on top of pull request #760 (Python 3.7 support).

### CODE REVIEW

File: assopy/admin.py [hunk 1]
```diff
@@ -221,7 +221,7 @@ def do_edit_invoices(self, request, queryset):
 
     def edit_invoices(self, request):
         try:
-            ids = map(int, request.GET['id'].split(','))
+            ids = list(map(int, request.GET['id'].split(',')))
```
Comment (artcz) [inline assopy/admin.py]: maybe we can replace that with list comprehension in the same PR?

File: assopy/admin.py [hunk 2]
```diff
@@ -723,16 +723,11 @@ def do_csv_invoices(self, request, queryset):
...
 
-        def e(d):
-            for k, v in d.items():
-                d[k] = v.encode('utf-8')
-            return d
-
         ofile = StringIO()
         writer = csv.DictWriter(ofile, fieldnames=columns)
-        writer.writerow(dict(zip(columns, columns)))
+        writer.writerow(dict(list(zip(columns, columns))))
```
Comment (artcz) [inline assopy/admin.py]: is that doing the same thing as `writer.writeheader()`?

File: assopy/management/commands/render_voucher.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ def handle(self, *args, **options):
         except IndexError:
             raise CommandError('codice conferenza non specificato')
 
-        oids = map(int, args[1:])
+        oids = list(map(int, args[1:]))
```
Comment (artcz) [inline assopy/management/commands/render_voucher.py]: same as above – maybe list comprehension instead?

File: assopy/models.py [hunk 4]
```diff
@@ -266,7 +266,7 @@ def create_from_profile(self, user, profile):
             birthday = profile.get('birthday', '').split('-')
             if birthday[0] == '0000':
                 birthday[0] = '1900'
-            identifier.birthday = date(*map(int, birthday))
+            identifier.birthday = date(*list(map(int, birthday)))
```
Comment (artcz) [inline assopy/models.py]: I was thinking about list comprehension here as well, but `strptime` sounds better.

File: p3/management/commands/get_attendees_csv.py [hunk 5]
```diff
@@ -62,17 +62,7 @@ def handle(self, *args, **options):
...
-        def utf8(d):
-            d = dict(d)
-            for k, v in d.items():
-                try:
-                    d[k] = v.encode('utf-8')
-                except:
-                    pass
-            return d
-
+        writer.writerow(dict(list(zip(columns, columns))))
```
Comment (artcz) [inline p3/management/commands/get_attendees_csv.py]: writeheader?

File: p3/utils.py [hunk 6]
```diff
@@ -97,7 +97,7 @@ def conference_ticket_badge(tickets):
         else:
             tagline = p3c.tagline
             experience = p3c.python_experience
-            tdays = map(lambda x: datetime.date(*map(int, x.split('-'))), filter(None, p3c.days.split(',')))
+            tdays = [datetime.date(*list(map(int, x.split('-')))) for x in [_f for _f in p3c.days.split(',') if _f]]
```
Comment (artcz) [inline p3/utils.py]: there's still one map() inside. :)

File: requirements.txt [hunk 7]
```diff
@@ -21,33 +21,34 @@ django-taggit==0.14.0
...
 requests==2.5.3
 responses==0.8.1
 requests-oauthlib==0.4.2
 simplejson==3.6.5
-six==1.9.0
-stripe==1.19.1
-suds==0.4
+six>1.9.0
+stripe
+suds-jurko
```
Comment (artcz) [inline requirements.txt:51]: what's suds-jurko?

Comment (fpierfed) [reply requirements.txt:51]: It is the Python3.7 compatible version of suds.

File: assopy/admin.py (hunk 2 above)
Comment (fpierfed) [reply assopy/admin.py]: yes

File: assopy/admin.py (hunk 1 above)
Comment (fpierfed) [reply assopy/admin.py]: yup

File: assopy/management/commands/render_voucher.py (hunk 3 above)
Comment (fpierfed) [reply assopy/management/commands/render_voucher.py]: ok

File: assopy/models.py (hunk 4 above)
Comment (fpierfed) [reply assopy/models.py]: done

File: p3/management/commands/get_attendees_csv.py (hunk 5 above)
Comment (fpierfed) [reply p3/management/commands/get_attendees_csv.py]: yes

File: p3/utils.py (hunk 6 above)
Comment (fpierfed) [reply p3/utils.py]: fixed

File: requirements.txt (hunk 7 above)
Comment (matrixise) [reply requirements.txt:51]: yep, but suds is for soap, and we don't support SOAP.

Comment (fpierfed) [reply requirements.txt:51]: It is used in assopy:

epcon ➤ rg -t py suds                                                                                                              misc/assopy-dist/setup.py
36:        'suds',

assopy/clients/vies.py
3:from suds.client import Client

Comment (artcz) [reply requirements.txt:51]: It was probably used long time ago, but it's not used right now, also I removed that whole veis "integration" few days ago -> https://github.com/EuroPython/epcon/pull/806

**Comment by umgelurgel** [comment]
Closing in favour of https://github.com/EuroPython/epcon/pull/851, migrating the project to python 3.6

**Pull request closed without merging**